src/hotspot/share/runtime/deoptimization.cpp

```diff
@@ -2465,6 +2465,7 @@ Deoptimization::UnrollBlock* Deoptimization::uncommon_trap(JavaThread* thread, j
     // This enters VM and may safepoint
     uncommon_trap_inner(thread, trap_request);
   }
+  HandleMark hm(thread);
```

Comment:
This is fine. I generally prefer the HandleMark closer to the lifetimes of the Handles that it protects.  If you put this HandleMark in fetch_unroll_info_helper() would it be needed before the line:
  Handle exceptionObject;
Or are there Handles created before there?
Given how complicated this code is, where you have the HandleMark is fine.  Well done finding this bug and writing a test for it!

Reply:
Thanks!

There are also Handles used in static method eliminate_allocations and eliminate_locks.